CONTRIBUTING.md

```diff
@@ -112,6 +112,28 @@ Siglens currently support `Linux` and `macOS` for development purposes.
    sudo yum install gcc g++ make
    ```
 
+## How to work with PEG
+
+To contribute to PEG related stories, you need to convert PEG files to Go files. 
+
+First, install pigeon using the following command:
+
+```bash
+go get -u github.com/mna/pigeon
+```
+
+Then, you can convert a PEG file to a Go file using the following command:
+
+```bash
+~/go/bin/pigeon -o ./pkg/ast/spl/spl.peg ./pkg/ast/spl/spl.go
+```
+
+Once a PEG file is converted into a Go file, you can then utilize the parsed variables in your subsequent code implementation.
+
+For more information on PEG syntax, you can refer to the [official PEG syntax documentation](https://pkg.go.dev/github.com/mna/pigeon#hdr-PEG_syntax).
+
+For a deeper understanding of PEG parsing, you can read this [series of articles](https://medium.com/@gvanrossum_83706/peg-parsing-series-de5d41b2ed60).
```

Comment:
Move these closer to the top of this section since people have to understand PEG before they can update the files. So the flow should be: learn PEG/pigeon, then update the .peg files, then compile the .peg files to .go files

Reply:
I just pushed a commit to re-order the instructions